### ISSUE TITLE
Exception 'KeyError' with recursive search

### ISSUE THREAD

**Issue by gregorywychowaniec-zt** [meeb/whoisit#40]
Hello,

I got some `KeyError: 'title'` exception in the new function `recursive_merge_lists` in `utils.py`. I add a try catch in my own development but it can happen to others.

**Comment by meeb** [comment]
Hi, can you post the full trace and what domain or entity you're looking up please?

**Comment by gregorywychowaniec-zt** [comment]
Sorry for the delay, i did some debug and i got the exception with the domains "showroomprive.com" , "infopro-digital.com" or "canalplus.com". 
When I inspect the debug, the call `recursive_merge_lists` is made with `depup_on="eventAction"` a first time without error but a second call is made with `depup_on="title"` with errors.
This list was the problem with the domain `showroomprive.com` : 
```
[
    {
        'description': ['Service subject to Terms of Use.'], 
        'links': [{'href': 'https://rdap.nameshield.net/rdap/terms', 'rel': 'alternate', 'type': 'text/html', 'value': 'https://rdap.nameshield.net/rdap/terms'}], 
        'title': 'RDAP Terms of Service'
    }, 
    {
        'description': ['Support page.'], 
        'links': [{'href': 'https://rdap.nameshield.net/rdap/support', 'rel': 'alternate', 'type': 'text/html', 'value': 'https://rdap.nameshield.net/rdap/support'}]
    }, 
    {
        'description': ['This response conforms to the RDAP Operational Profile for gTLD Registries and Registrars version 1.0']
    }, 
    {
        'description': ['For more information on domain status codes, please visit https://icann.org/epp'], 
        'links': {'href': 'https://icann.org/epp', 'rel': 'alternate', 'type': 'text/html', 'value': 'https://icann.org/epp'}, 
        'title': 'Status Codes'
    }, 
    {
        'description': ['URL of the ICANN RDDS Inaccuracy Complaint Form: https://www.icann.org/wicf'], 
        'links': {'href': 'https://www.icann.org/wicf', 'rel': 'alternate', 'type': 'text/html', 'value': 'https://www.icann.org/wicf'}, 
        'title': 'RDDS Inaccuracy Complaint Form'
    }
]
```
As you can see, there are entries without title key.

**Comment by meeb** [comment]
The above commit should resolve this, I'll bundle it into the next release.

**Comment by meeb** [comment]
This is fixed in the 3.0.4 release available now.
